app/migrations/Version20150531210410.php

```diff
@@ -0,0 +1,109 @@
+
+        // associate users with customers by email
+        $this->addSql('ALTER TABLE sylius_user ADD customer_id INT NOT NULL');
+        $this->addSql('UPDATE sylius_user INNER JOIN sylius_customer ON sylius_user.email = sylius_customer.email SET sylius_user.customer_id = sylius_customer.id');
+        $this->addSql('ALTER TABLE sylius_user ADD CONSTRAINT FK_569A33C09395C3F3 FOREIGN KEY (customer_id) REFERENCES sylius_customer (id)');
+        $this->addSql('CREATE UNIQUE INDEX UNIQ_569A33C09395C3F3 ON sylius_user (customer_id)');
+
+        // migrate groups
+        $this->addSql('INSERT INTO sylius_customer_group (customer_id, group_id) SELECT sylius_customer.id, sylius_user_group.group_id FROM sylius_user_group INNER JOIN sylius_user ON sylius_user_group.user_id = sylius_user.id INNER JOIN sylius_customer ON sylius_user.customer_id = sylius_customer.id');
+        $this->addSql('DROP TABLE sylius_user_group');
+        $this->addSql('DROP INDEX UNIQ_F97F76A45E237E06 ON sylius_group');
+        $this->addSql('ALTER TABLE sylius_group DROP roles');
+
+        // migrate addresses
+        $this->addSql('ALTER TABLE sylius_address DROP FOREIGN KEY FK_B97FF058A76ED395');
+        $this->addSql('DROP INDEX IDX_B97FF058A76ED395 ON sylius_address');
+        $this->addSql('ALTER TABLE sylius_address ADD customer_id INT DEFAULT NULL');
+        $this->addSql('ALTER TABLE sylius_address ADD CONSTRAINT FK_B97FF0589395C3F3 FOREIGN KEY (customer_id) REFERENCES sylius_customer (id) ON DELETE SET NULL');
+        $this->addSql('CREATE INDEX IDX_B97FF0589395C3F3 ON sylius_address (customer_id)');
+        // associate addresses with customers
+        $this->addSql('UPDATE sylius_address INNER JOIN sylius_user ON sylius_address.user_id = sylius_user.id INNER JOIN sylius_customer ON sylius_user.customer_id = sylius_customer.id SET sylius_address.customer_id = sylius_customer.id');
+        // drop relation to user
+        $this->addSql('ALTER TABLE sylius_address DROP user_id');
+
+        $this->addSql('ALTER TABLE sylius_promotion_coupon CHANGE per_user_usage_limit per_customer_usage_limit INT DEFAULT NULL');
+
+        // migrate orders
+        $this->addSql('ALTER TABLE sylius_order DROP FOREIGN KEY FK_6196A1F9A76ED395');
+        $this->addSql('DROP INDEX IDX_6196A1F9A76ED395 ON sylius_order');
+        $this->addSql('ALTER TABLE sylius_order ADD customer_id INT DEFAULT NULL');
+        $this->addSql('ALTER TABLE sylius_order ADD CONSTRAINT FK_6196A1F99395C3F3 FOREIGN KEY (customer_id) REFERENCES sylius_customer (id)');
+        $this->addSql('CREATE INDEX IDX_6196A1F99395C3F3 ON sylius_order (customer_id)');
+        // create customers based on guest checkouts
+        $this->addSql('INSERT INTO sylius_customer (email, email_canonical, created_at) SELECT sylius_order.email, sylius_order.email, sylius_order.created_at FROM sylius_order LEFT JOIN sylius_customer ON sylius_order.email = sylius_customer.email WHERE sylius_customer.email is NULL');
```

Comment:
The problem is that the email address cannot be null in sylius_customer table so you will have the following violation if you keep it as it is

```
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'email' cannot be null  
```